docs/frontend/tensorflow.rst

```diff
@@ -0,0 +1,226 @@
+
+.. _TFParser: https://github.com/dmlc/tvm/blob/77445311540c0dfa7b124304b5cf89da6f2c210f/python/tvm/relay/frontend/tensorflow_parser.py
+
+The model should be exported with a number of transformations to prepare the model for inference. It is also important to set ```add_shapes=True```, as this will embed the output shapes of each node into the graph. Here is one function to export a model as a protobuf given a session:
+
+.. code:: python
+
+    import tensorflow as tf
+    from tensorflow.tools.graph_transforms import TransformGraph
+
+    def export_pb(session):
+        with tf.gfile.GFile("myexportedmodel.pb", "wb") as f:
+            inputs = ["myinput1", "myinput2"] # replace with your input names
+            outputs = ["myoutput1"] # replace with your output names
+            graph_def = session.graph.as_graph_def(add_shapes=True)
+            graph_def = tf.graph.util.convert_variables_to_constants(session, graph_def, outputs)
+            graph_def = TransformGraph(
+                graph_def,
+                inputs,
+                outputs,
+                [
+                    "remove_nodes(op=Identity, op=CheckNumerics, op=StopGradient)",
+                    "sort_by_execution_order", # sort by execution order after each transform to ensure correct node ordering
+                    "remove_device",
+                    "sort_by_execution_order",
+                    "fold_batch_norms",
+                    "sort_by_execution_order",
+                    "fold_old_batch_norms",
+                    "sort_by_execution_order"
+                ]
+            )
+            f.write(graph_def.SerializeToString())
+
+Another method is to `export and freeze the graph <https://github.com/tensorflow/models/tree/master/research/slim#exporting-the-inference-graph>`_.
+
+Import the Model
+----------------
+
+Explicit Shape:
+~~~~~~~~~~~~~~~
+
+To ensure shapes can be known throughout the entire graph, pass the ```shape``` argument to ```from_tensorflow```. This dictionary maps input names to input shapes. Please refer to these `test cases <https://github.com/dmlc/tvm/blob/master/nnvm/tests/python/frontend/tensorflow/test_forward.py#L36>`_ for examples.
+
+Data Layout
+~~~~~~~~~~~
+
+Most TensorFlow models are released with NHWC layout. NCHW layout often provides better performance, especially on GPU. The TensorFlow frontend can automatically convert the model's data layout by passing the argument ```layout='NCHW'``` to ```from_tensorflow```.
+
+Best Practices
+--------------
+
+- Use static tensor shapes instead of dynamic shapes (remove ```None``` dimensions).
+- Use static RNN instead of dynamic RNN, as ```TensorArray``` isn't supported yet.
+
+Supported Ops
+-------------
+
+- Abs
```

Comment:
Maybe, we can automatically generate this by reading the map in the parser? :)

Reply:
That's an interesting idea :) how would we plug it into this file?

Reply:
I actually think, for now, it's reasonable to ask people to update this list when they add operator support to the frontend. When adding a new backend op, we always ask that they update the necessary docs there.

Reply:
Some extra supported control flow ops:  'Merge', 'Switch', 'NextIteration', 'Exit', 'Enter', 'LoopCond' 

It would be super helpful if we can plug the list into this file automatically, not sure how to do that. @tqchen @wweic